src/Core/Domain/Shipment/Command/AddShipmentCommand.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License version 3.0
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License version 3.0
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Shipment\Command;
+
+use DateTime;
+
+/**
+ * Add shipment.
+ */
+class AddShipmentCommand
+{
+    public function __construct(
+        private readonly int $orderId,
+        private readonly int $carrierId,
+        private readonly int $addressId,
+        private readonly float $shippingCostTaxExcluded,
+        private readonly float $shippingCostTaxIncluded,
+        private readonly array $products,
+        private readonly ?string $trakingNumber,
+        private readonly ?DateTime $packedAt = null,
+        private readonly ?DateTime $shippedAt = null,
+        private readonly ?DateTime $deliveredAt = null,
+        private readonly ?DateTime $cancelledAt = null
+    ) {
+    }
+
+    public function getOrderId(): int
+    {
+        return $this->orderId;
+    }
+
+    public function getCarrierId(): int
+    {
+        return $this->carrierId;
+    }
+
+    public function getAddressId(): int
+    {
+        return $this->addressId;
+    }
```

Comment:
The input of the command should be scalar values, so integers are perfect
But the stored value then should be a Value Object, so they should be transformed into `OrderId` `CarrierId` and `AddressId`